### ISSUE TITLE
OSC-Interface - SpatGris

### ISSUE THREAD

**Issue by delmenhorst** [GRIS-UdeM/SpatGRIS#328]
Hello,

I am on a M1 machine (MacOS Monterey) and using the latest Release (3.1.2) of SpatGris. I am about to send OSC-messages from Pure Data (0.52.1) to SpatGris. I can not figure out why it doesn't work (anymore). I cannot translate my old patches to the new osc system. 

I tried several experiments to get the bug. But I am not able to find the problem. First I thought it is about that I am sendung wrong value types. I seems not.
The example MAX patch works proberly, but when I am sending data form pure data to max and then to SpatGris: nothing happens, too.

Its very strange.  Maybe someone have an example patch for PD?
It did the same thing last year without problems.

I also have to mention that I cannot change the port number in SpatGris.

**Comment by delmenhorst** [comment]
Here is a test-patch:
[spatgris-osc-test.pd.zip](https://github.com/GRIS-UdeM/SpatGRIS/files/7776316/spatgris-osc-test.pd.zip)

**Comment by SamuelBeland** [comment]
It seems that pd is sending every numbers as if they were floating points, even values that are supposed to be integers (like source indexes). Because of this, SpatGRIS considers the messages to be ill-formed and simply discards them.

SpatGRIS should be more permissive when dealing with incoming data. Also, the OSC monitor in SpatGRIS should print helpful error messages when something gets discarded.

I also can't seem to be able to change the OSC port...

This is going to be fixed in the next release.

Thank you for reporting @delmenhorst.

**Comment by delmenhorst** [comment]
I assumed that, but I didn't find a way to change the values to integers. It seems not really possible. (for now and for me :)
Since my patch has worked in the past with an older Version of SpatGris, I don't know what I really could change?

Is the idea that the new release can handle the messages in this way? Or should I look for solutions in pd?

**Comment by SamuelBeland** [comment]
We are going to make it so that SpatGRIS can handle your current patch without modifications.

**Comment by delmenhorst** [comment]
C'est parfait. :)

**Comment by Normandeau** [comment]
V313
Apparemment c'est réglé. Est-ce que delmenhorst pourrait tester puisque vous avez ouvert cette issue. Merci.

**Comment by delmenhorst** [comment]
I've tested the changes.
Spatgris recognizes the OSC messages without the specific value formats. However, it no longer recognizes the polar coordinates.

<img width="1409" alt="Bildschirmfoto 2022-01-05 um 16 34 12" src="https://user-images.githubusercontent.com/39628224/148245153-17514200-15f8-421f-9aa7-be1b08c28610.png">


It is now also possible to change the port number.

Also, I have now found a way to format values ​​in Pure Data. But I think the current changes are still beneficial.
